File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        j={'I':1 , 'V':5 , 'X':10 , 'L':50 , 'C':100 ,'D':500, 'M':1000}
        ans = 0
        i=0
        while i < len(s):
            if i+1 <len(s):
                if s[i]=='I' and s[i+1]=='V':
                    ans += 4
                    i +=2
                    continue
                elif s[i]=='I' and s[i+1]=='X':
                    ans += 9
                    i +=2
                    continue    
                elif s[i]=='X' and s[i+1]=='L':
                    ans += 40
                    i+=2
                    continue       
                elif s[i]=='X' and s[i+1]=='C':
                    ans += 90
                    i +=2
                    continue     
                elif s[i]=='C' and s[i+1]=='D':
                    ans += 400
                    i+=2
                    continue 
                elif s[i]=='C' and s[i+1]=='M':
                    ans += 900
                    i+=2
                    continue 
            
            ans += j[s[i]]
            i+=1
        
        return ans

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
    Map<Character , Integer> numbers = new HashMap<>();
    numbers.put('I',1);
    numbers.put('V',5);
    numbers.put('X',10);
    numbers.put('L',50);
    numbers.put('C',100);
    numbers.put('D',500);
    numbers.put('M',1000);
    int ans= 0 ;
    for(int i = 0 ; i < s.length() ; i++){
     if(i+1<s.length()){
       if(s.charAt(i)=='I'&&s.charAt(i+1)=='V'){
        ans+= 4 ;
        i++;
        continue;
       }
       
       else if(s.charAt(i)=='I'&&s.charAt(i+1)=='X'){
        ans+= 9 ;
        i++;
        continue;
       }
        
        else if(s.charAt(i)=='X'&&s.charAt(i+1)=='L'){
        ans+= 40 ;
        i++;
        continue;
       }

        else if(s.charAt(i)=='X'&&s.charAt(i+1)=='C'){
        ans+= 90 ;
        i++;
        continue;
       }

        else if(s.charAt(i)=='C'&&s.charAt(i+1)=='D'){
        ans+= 400 ;
        i++;
        continue;
       }

        else if(s.charAt(i)=='C'&&s.charAt(i+1)=='M'){
        ans+= 900 ;
        i++;
        continue;
       }
     }
    
       ans+= numbers.get(s.charAt(i));
       } 
    return ans;             
    }
}